src/kbc_modules/eaa_kbc/mod.rs

```diff
@@ -154,7 +173,77 @@ impl EAAKbc {
         }
     }
 
-    fn kbs_trans_and_recv(&mut self, trans_data: &[u8], error_info: &str) -> Result<String> {
+    fn kbs_get_resource(&mut self, description: String) -> Result<Vec<u8>> {
+        let desc: ResourceDescription =
+            serde_json::from_str::<ResourceDescription>(description.as_str())?;
+
+        let (request, resource_info) = match desc.name.as_str() {
+            POLICY_RESOURCE_NAME => (
+                GetResourceRequest::new(GET_POLICY_COMMAND, desc.optional.clone()),
+                self.kbs_get_resource_info(desc.name.as_str())?,
+            ),
+            SIGSTORE_CONFIG_RESOURCE_NAME => (
+                GetResourceRequest::new(GET_SIGSTORE_CONFIG_COMMAND, desc.optional.clone()),
+                self.kbs_get_resource_info(desc.name.as_str())?,
+            ),
+            GPG_KEY_RING_RESOURCE_NAME => (
+                GetResourceRequest::new(GET_GPG_KEY_RING_COMMAND, desc.optional.clone()),
+                self.kbs_get_resource_info(desc.name.as_str())?,
+            ),
+            _ => return Err(anyhow!("This resource is not supported.")),
+        };
+
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+        let buffer_size: usize = resource_info.base64size.parse::<usize>()?;
+
+        let recv_string: String =
+            self.kbs_trans_and_recv(trans_data, buffer_size, "Get Resource")?;
+
+        let data_bytes = base64::decode(recv_string)?;
+
+        if data_bytes == b"Error".to_vec() {
+            return Err(anyhow!(
+                "Verdictd: The resource data is null, something failed."
+            ));
+        }
+
+        Ok(data_bytes)
+    }
+
+    fn kbs_get_resource_info(&mut self, resource_name: &str) -> Result<ResourceInfo> {
+        let request = GetResourceInfoReq::new(resource_name);
+        let trans_json = serde_json::to_string(&request)?;
+        let trans_data: &[u8] = trans_json.as_bytes();
+
+        let recv_string: String =
+            self.kbs_trans_and_recv(trans_data, DEFAULT_RECV_SIZE, "Get Resource Info")?;
+
+        let response = serde_json::from_str::<GetResourceInfoResponse>(recv_string.as_str())?;
+
+        match response.status.as_str() {
+            "OK" => {
+                if let Some(info) = response.data {
+                    Ok(info)
+                } else {
+                    return Err(anyhow!("Verdictd: Resource info payload is null!"));
+                }
+            }
+            "Fail" => return Err(anyhow!(format!("Verdictd: {}", response.error.unwrap()))),
+            _ => {
+                return Err(anyhow!(
+                    "Can't understand the GetResourceInfoResponse status!"
```

Comment:
- Personally, I wouldn't include exclamations in error messages: they don't add any additional information. It's going to be clear to the admin this is an error so that is sufficient.
- I'd also avoid apostrophe's where possible to since these signify a contraction and not all reads will understand these shortened forms of words. Plus they can cause problems for log parsers / formatters.